botorch/acquisition/multi_objective/utils.py

```diff
@@ -161,3 +174,224 @@ def prune_inferior_points_multi_objective(
+    optimal_outputs = torch.tensor([], **tkwargs)
+    num_tries = 0
+    ratio = 2
+    while ratio > 1 and num_tries < max_tries:
+        X = draw_sobol_samples(bounds=bounds, n=pop_size, q=1).squeeze(-2)
+        Y = model.posterior(X).mean
+        X_aug = torch.cat([optimal_inputs, X], dim=0)
+        Y_aug = torch.cat([optimal_outputs, Y], dim=0)
+        pareto_mask = is_non_dominated(weight * Y_aug)
+        optimal_inputs = X_aug[pareto_mask]
+        optimal_outputs = Y_aug[pareto_mask]
+        num_found = len(optimal_inputs)
+        ratio = ceil(num_points / num_found)
+        num_tries = num_tries + 1
+    # If maximum number of retries exceeded throw out a runtime error.
+    if ratio > 1:
+        error_text = f"Only found {num_found} optimal points instead of {num_points}."
+        raise RuntimeError(error_text)
+    else:
+        return optimal_inputs[:num_points], optimal_outputs[:num_points]
+
+
+def sample_optimal_points(
+    model: Model,
+    bounds: Tensor,
+    num_samples: int,
+    num_points: int,
+    optimizer: Callable[
+        [GenericDeterministicModel, Tensor, int, bool, Any], Tuple[Tensor, Tensor]
+    ] = random_search_optimizer,
+    num_rff_features: int = 512,
+    maximize: bool = True,
+    optimizer_kwargs: dict = None,
+) -> Tuple[Tensor, Tensor]:
+    r"""Compute a collection of optimal inputs and outputs from samples of a Gaussian
+    Process (GP).
+
+    Steps:
+    (1) The samples are generated using random Fourier features (RFFs).
+    (2) The samples are optimized sequentially using an optimizer.
+
+    TODO: We can generalize the GP sampling step to accommodate for other sampling
+        strategies rather than restricting to RFFs e.g. decoupled sampling.
+
+    Args:
+        model: The model. This does not support models which include fantasy
+            observations.
+        bounds: A `2 x d`-dim Tensor containing the input bounds.
+        num_samples: The number of GP samples.
+        num_points: The number of optimal points to be outputted.
+        optimizer: A callable that solves the deterministic optimization problem.
+        num_rff_features: The number of random Fourier features.
+        maximize: If true, we consider a maximization problem.
+        optimizer_kwargs: The additional arguments for the optimizer.
+
+    Returns:
+        A two-element tuple containing
+
+        - A `num_samples x num_points x d`-dim Tensor containing the collection of
+            optimal inputs.
+        - A `num_samples x num_points x M`-dim Tensor containing the collection of
+            optimal objectives.
+    """
+    tkwargs = {"dtype": bounds.dtype, "device": bounds.device}
+    M = model.num_outputs
+    d = bounds.shape[-1]
+    if M == 1:
+        if num_points > 1:
+            raise UnsupportedError(
+                "For single-objective optimization `num_points` should be 1."
+            )
+    if optimizer_kwargs is None:
+        optimizer_kwargs = dict()
```

Comment:
```suggestion
        optimizer_kwargs = {}
```
A Meta-internal linter prefers this (C408 from flake8-comprehensions, if anyone is curious https://pypi.org/project/flake8-comprehensions/)